src/error/GraphQLError.ts

```diff
@@ -121,8 +122,10 @@ export class GraphQLError extends Error {
     positions?: Maybe<ReadonlyArray<number>>,
     path?: Maybe<ReadonlyArray<string | number>>,
     originalError?: Maybe<Error & { readonly extensions?: unknown }>,
+    // eslint-disable-next-line @typescript-eslint/unified-signatures
     extensions?: Maybe<GraphQLErrorExtensions>,
   );
+  // eslint-disable-next-line @typescript-eslint/unified-signatures
   constructor(message: string, args?: GraphQLErrorArgs);
   constructor(message: string, ...rawArgs: BackwardsCompatibleArgs) {
```

Comment:
had to do this so the following is not highlighted as deprecated.

```ts
new GraphQLError("foo")
```

new ~~GraphQLError~~("foo")

seems like `@typescript-eslint/unified-signatures` does not work well with `@deprecated`

Reply:
I propose to solve this issue in a different way in #3514

Reply:
rebased on top of your change.